### PR TITLE
Add x-mitre-data-component

### PR DESCRIPTION
**Pull request by ZikyHD**
Add :
- data_component_stix_mapping
- get_enterprise_data_components()
- get_data_components() (maybe in the future)
- get_techniques_detected_by_all_data_components()
- get_techniques_detected_by_data_component()
- get_data_component_by_technique()

### CODE REVIEW

**Comment by Cyb3rWard0g** [comment]
Once again @ZikyHD ! Thank you very much for taking the time to do this! It looks good to me just by going through the code. I have to test your branch on my dev vm. I will be doing that on Friday this week. I will merge after that. 

Thank you again! 🍻

**Comment by ZikyHD** [comment]
@Cyb3rWard0g any update on my proposal ? 😅

**Comment by Cyb3rPandaH** [comment]
Hey @ZikyHD , hope everything is going well.

Sorry for the late reply, @Cyb3rWard0g and I are reviewing the PR. We will merge it over the weekend 👍 

Thank you again for your contribution guys 🙏

**Comment by Cyb3rPandaH** [comment]
Quick question @ZikyHD 

What is the idea behind the function **get_techniques_detected_by_all_data_components** (Line 1479). Is it to get all techniques that have data components?

Thanks,

Jose

File: attackcti/attack_api.py [hunk 1]
```diff
@@ -839,7 +867,25 @@ def get_mitigations(self, stix_format=True):
...
         return enterprise_mitigations
-    
+
+    def get_data_components(self, stix_format=True):
+        """ Extracts all the available data components STIX objects across all ATT&CK matrices
+        Args:
+            stix_format (bool):  Returns results in original STIX format or friendly syntax (i.e. 'attack-pattern' or 'technique')
+        """
+        enterprise_data_components = self.get_enterprise_data_components()
+        mobile_data_components = self.get_mobile_data_components()
```
Comment (Cyb3rWard0g) [inline attackcti/attack_api.py:877]: missing get_mobile_data_components()

File: attackcti/attack_api.py [hunk 2]
```diff
@@ -839,7 +867,25 @@ def get_mitigations(self, stix_format=True):
...
+        ics_data_components = self.get_ics_data_components()
```
Comment (Cyb3rWard0g) [inline attackcti/attack_api.py:878]: missing get_ics_data_components()

**Review by Cyb3rWard0g — CHANGES_REQUESTED** [review]
Hello @ZikyHD ! Thank you again! @Cyb3rPandaH  and I are working on it at the moment. These two functions are missing. I was wondering if maybe the code was not committed as part of the PR or maybe data components do not exist yet in ICS and mobile data. We are going to check that out that too right now.

File: attackcti/attack_api.py (hunk 2 above)
Comment (Cyb3rWard0g) [reply attackcti/attack_api.py:878]: just validated that there are not ICS data components.

File: attackcti/attack_api.py (hunk 1 above)
Comment (Cyb3rWard0g) [reply attackcti/attack_api.py:877]: just validated that there are not data components for mobile

**Comment by Cyb3rWard0g** [comment]
Thank you @ZikyHD ! I will accept the PR and comment out the functions that do not retrieve data components from Mobile and ICS inside of get_data_components(). 

Thank you again for the contribution and your patience! We really appreciate it!

**Pull request merged**